src/main/java/com/taboola/backstage/internal/extensions/BackstageExtensionUtilsImpl.java

```diff
@@ -0,0 +1,21 @@
+package com.taboola.backstage.internal.extensions;
+
+/**
+ * Created by vladi
+ * Date: 3/4/2018
+ * Time: 12:37 AM
+ * By Taboola
+ */
+public class BackstageExtensionUtilsImpl implements BackstageExtensionUtils {
```

Comment:
why is it utils ? currently it has only one method which return the endpoint factory, clearly not "utils".
I suggest to rename it to EndpointFactoryProvider/Creator or such... if there is a need to add another method to here in the future - we either will change the name of the class to be more global (but not utils), or add it to a new class.

Reply:
@barkamir I dont want to change often interfaces to be more backward compatible (as much as possible). But surely we can rename it to something more clearer.